tribits/core/utils/TribitsDeprecatedHelpers.cmake

```diff
@@ -0,0 +1,105 @@
+# met:
+#
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the Corporation nor the names of the
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# ************************************************************************
+# @HEADER
+
+include(MessageWrapper)
+include(TribitsParseArgumentsHelpers)
+
+
+# @FUNCTION: tribits_deprecated()
+#
+# Notify the user that some TriBITS functionality is deprecated. Depending on
+# the value of TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE, this can do one of
+# several things:
+#
+# - ``DEFAULT`` or empty string: issue a CMake ``DEPRECATION`` message and
+#   continue.
+# - ``AUTHOR_WARNING``: issue a CMake ``AUTHOR_WARNING`` message and continue.
+# - ``SEND_ERROR``: issue a CMake ``SEND_ERROR`` message and continue.
+# - ``FATAL_ERROR``: issue a CMake ``FATAL_ERROR`` message and exit.
+# - ``IGNORE``: issue no message and continue.
+#
+# Usage::
+#
+#   tribits_deprecated(<message>)
+#
+function(tribits_deprecated  message)
+  if ("${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}" STREQUAL "" OR
+    "${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}" STREQUAL "DEFAULT"
+    )
+    message_wrapper(DEPRECATION "${message}")
+  elseif ("${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}" STREQUAL "AUTHOR_WARNING")
+    message_wrapper(AUTHOR_WARNING "${message}")
+  elseif ("${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}" STREQUAL "SEND_ERROR")
+    message_wrapper(SEND_ERROR "${message}")
+  elseif ("${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}" STREQUAL "FATAL_ERROR")
+    message_wrapper(FATAL_ERROR "${message}")
+  elseif (NOT "${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}" STREQUAL "IGNORE")
+    message_wrapper(FATAL_ERROR "Invalid value for TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE: '${TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE}'")
+  endif()
```

Comment:
As it stands right now, empty string and `DEFAULT` cause a `DEPRECATION` message to be printed. `DEPRECATION` is not actually a valid value for `TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE`. Are you proposing changing this design?

Reply:
Just replacing the value `DEFAULT` with `DEPRECATION`.   Since the default value for `TRIBITS_HANDLE_TRIBITS_DEPRECATED_CODE` will be `DEPRECATION`, the user will not have to type anything to get the default value.  And since this is the mode passed to `message()`, people familiar with CMake should know what that means.  Only the value of `IGNORE` needs to be explained (because it avoids the `message()` command being called at all).